### ISSUE TITLE
[BUG][JS] JSDoc does not indicate whether a parameter is required or not

### ISSUE THREAD

**Issue by kuramen** [OpenAPITools/openapi-generator#15014]
##### Description

When generating an SDK JS using OpenAPI Generator, the JSDoc generated does not indicate whether an element is required or not. In JSDoc, required parameters should be indicated with brackets, like this: `[requiredParam]`.

##### openapi-generator version

I'm using version 6.2.0 of OpenAPI Generator. It' not a regression.

##### OpenAPI declaration file content or url

Here is an example of the OpenAPI JSON file I'm using :
```
{
  "openapi": "3.0.0",
  "info": {
    "title": "My API",
    "version": "1.0.0"
  },
  "paths": {
    "/users/{userId}": {
      "get": {
        "summary": "Get a user by ID",
        "operationId": "getUserById",
        "parameters": [
          {
            "name": "userId",
            "in": "path",
            "description": "The ID of the user to get",
            "required": true,
            "schema": {
              "type": "integer"
            }
          },
          {
            "name": "email",
            "in": "query",
            "description": "The email of the user to get",
            "required": false,
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Successful response",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "id": {
                      "type": "integer"
                    },
                    "name": {
                      "type": "string"
                    },
                    "email": {
                      "type": "string"
                    }
                  },
                  "required": [
                    "id",
                    "name"
                  ]
                }
              }
            }
          }
        }
      }
    }
  }
}
```

##### Generation Details

I'm using the following command to generate the SDK JS:

`openapi-generator-cli generate -i openapi.json -g javascript -o sdk-js`

##### Steps to reproduce

1. Create an OpenAPI JSON file with required and optional parameters.
2. Run the above command to generate an SDK JS from the OpenAPI JSON file.
3. Observe that the JSDoc generated for the SDK JS does not indicate whether a parameter is required or not.

##### Related issues/PRs

I have searched the issues and PRs on the OpenAPI Generator repository, but have not found any related to this specific issue.

##### Actual output vs expected output

Actual output:

```
/**
 * @param {number} userId The ID of the user to get
 * @param {string} email The email of the user to get
 */
```

Expected output:

```
/**
 * @param {number} userId The ID of the user to get
 * @param {string} [email] The email of the user to get
 */
```

As you can see, the expected output includes the [required] indicator for the userId parameter, indicating that it is a required parameter, while the email parameter is optional and does not have the [required] indicator.

This behavior is important for developers using the SDK JS, as it helps them understand the expected behavior of the API endpoint and avoid errors caused by missing or incorrect parameter values.

I hope this helps clarify the issue. Let me know if you have any further questions or concerns.

**Comment by wing328** [comment]
@kuramen thanks for reporting the issue. I've filed https://github.com/OpenAPITools/openapi-generator/pull/15021 to fix it.

can you please review to see if that's what you're looking for?

**Comment by wing328** [comment]
fixed via #15032, please pull the latest master to give it a try.
